r-package/grf/tests/testthat/test_input_utilities.R

```diff
@@ -0,0 +1,28 @@
+library(grf)
+
+test_that("forests handle matrix and data.frame inputs equally", {
+  
+  n = 100
+  p = 2
+  Xm = matrix(rnorm(n*p), n, p)
+  Xd = as.data.frame(Xm)
+  Y = rnorm(n)
+  W = rbinom(0.5, size=1, n=n)
+  Z = runif(n)
+  
+  # Matrix input
+  rfm = regression_forest(Xm, Y, num.trees=5, seed=1234)
```

Comment:
I was thinking we could add a unit (not integration) test, and just check the output of `create_data_matrices` directly instead of training forests. I'm hoping to move more towards unit tests, to make debugging test failures easier and cut down on the long R build times we've been seeing.

Reply:
Makes sense! I'll update this.